src/isar/robot/robot.py

```diff
@@ -151,61 +171,115 @@ def _pause_mission_done_handler(self) -> None:
             self.pause_mission_thread = None
 
             if error_message:
-                self.robot_service_events.mission_failed_to_pause.trigger_event(
+                self.robot_service_to_state_machine_events.mission_failed_to_pause.trigger_event(
                     error_message
                 )
             else:
-                self.robot_service_events.mission_successfully_paused.trigger_event(
+                self.robot_service_to_state_machine_events.mission_successfully_paused.trigger_event(
                     True
                 )
 
     def _start_mission_event_handler(self, event: Event[Mission]) -> None:
-        start_mission = event.consume_event()
-        if start_mission is not None:
-            if (
-                self.start_mission_thread is not None
-                and self.start_mission_thread.is_alive()
-            ):
-                self.logger.warning(
-                    "Attempted to start mission while another mission was starting."
-                )
-                self.start_mission_thread.join()
+        mission = event.consume_event()
+        if mission is None:
+            return
+        if (
+            self.prepare_mission_thread is not None
+            and self.prepare_mission_thread.is_alive()
+        ):
+            self.logger.warning(
+                "Attempted to prepare mission while another mission was preparing."
+            )
+            self.robot_service_to_robot_service_events.cancel_mission_preparation.trigger_event(
+                True
+            )
+            self.prepare_mission_thread.join()
+        if (
+            self.start_mission_thread is not None
+            and self.start_mission_thread.is_alive()
+        ):
+            self.logger.warning(
+                "Attempted to prepare mission while another mission was starting."
+            )
+            self.robot_service_to_robot_service_events.cancel_mission_start.trigger_event(
+                True
+            )
+            self.start_mission_thread.join()
```

Comment:
This would block the whole robot.py thread. See other comments for more details, but the gist of it is that this shouldn't be possible if we always wait for a thread to complete before reporting the output.